hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/event/remote/TargetDeletedEvent.java

```diff
@@ -26,20 +28,33 @@ public TargetDeletedEvent() {
     }
 
     /**
-     * Constructor for json serialization.
-     * 
+     *
      * @param tenant
      *            the tenant
      * @param entityId
      *            the entity id
+     * @param controllerId
+     *            the controllerId of the target
+     * @param targetAddress
+     *            the target address
      * @param entityClass
      *            the entity class
      * @param applicationId
      *            the origin application id
      */
-    public TargetDeletedEvent(final String tenant, final Long entityId, final String entityClass,
-            final String applicationId) {
+    public TargetDeletedEvent(final String tenant, final Long entityId, final String controllerId,
+            final String targetAddress, final String entityClass, final String applicationId) {
```

Comment:
We need the class here, because we have to check, if a UI event could match with a RemoteEvent. If both events matches, then we can skip the processing for this event e.g. for the notification counter
See `BaseUIEntityEvent#matchRemoteEvent`